libbeat/beat/beat.go

```diff
@@ -65,15 +67,36 @@ func NewBeat(name string, version string, bt Beater) *Beat {
 		Name:    name,
 		BT:      bt,
 		UUID:    uuid.NewV4(),
+
+		done: make(chan struct{}),
 	}
 
 	return &b
 }
 
 // Initiates and runs a new beat object
-func Run(name string, version string, bt Beater) *Beat {
+func Run(name string, version string, bt Beater) {
 	b := NewBeat(name, version, bt)
 
+	// Runs beat inside a go process
+	go func() {
+		logp.Info("Start Beat")
+		b.Start()
+		b.Stop()
```

Comment:
is it safe to call 'Stop' like twice? I think calling Stop twice used to be a problem in the past.